lib/services/fanout_test.go

```diff
@@ -70,3 +71,71 @@ func TestFanoutInit(t *testing.T) {
 	default:
 	}
 }
+
+/*
+goos: linux
+goarch: amd64
+pkg: github.com/gravitational/teleport/lib/services
+cpu: Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
+BenchmarkFanoutRegistration-16       	       1	118856478045 ns/op
+*/
+// NOTE: this benchmark exists primarily to "contrast" with the set registration
+// benchmark below, and demonstrate why the set-based strategy is necessary.
+func BenchmarkFanoutRegistration(b *testing.B) {
+	const iterations = 100_000
+
+	for n := 0; n < b.N; n++ {
+		f := NewFanout()
+		f.SetInit()
+
+		var wg sync.WaitGroup
+
+		for i := 0; i < iterations; i++ {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				w, err := f.NewWatcher(context.TODO(), types.Watch{
+					Name:  "test",
+					Kinds: []types.WatchKind{{Name: "spam"}, {Name: "eggs"}},
+				})
+				require.NoError(b, err)
+				defer w.Close()
```

Comment:
nit: drop the defer? Doesn't make a lot of sense at this point.

```suggestion
				w.Close()
```